_rules/SC2-2-1+SC3-2-5-meta-refresh.md

```diff
@@ -1,77 +1,185 @@
-|-------------|----------
-| type        | TestResult
-| outcome     | Passed
-| description |
-
-### step1-fail
-
-| Property    | Value
-|-------------|----------
-| type        | TestResult
-| outcome     | Failed
-| description | Meta refresh should not be used unless it is instantaneous.
+### Applicability
+
+The rule applies to the first [valid](https://www.w3.org/TR/html/document-metadata.html#statedef-http-equiv-refresh) `<meta http-equiv="refresh">` element with a `content` attribute in a document.
+
+### Expectation
+
+The `time` of the `content` attribute is not greater than 0.
+
+**Note**: See [Refresh state (`http-equiv="refresh"`)](https://www.w3.org/TR/html/document-metadata.html#statedef-http-equiv-refresh) for a precise description on how to determine the `time`.
+
+## Assumptions  
+
+* This test assumes no functionality was provided by the website for the user to adjust the timer. 
+* This test assumes that the refresh was not [essential](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-required-behaviors.html#essentialdef), which is listed as a valid exception to SC 2.2.1.
+
+## Accessibility Support 
+
+There are no major accessibility support issues known for this rule.
+
+## Background  
+
+- H76: Using meta refresh to create an instant client-side redirect
+- F40: Failure of Success Criterion 2.2.1 and 2.2.4 due to using meta redirect with a time limit
+- F41: Failure of Success Criterion 2.2.1, 2.2.4, and 3.2.5 due to using meta refresh with a time-out
+
+## Test Cases
+
+### Passed
+
+```html
+<!-- redirects immediately -->
+  <head>           
+    <meta http-equiv="refresh" content="0; URL='https://auto-wcag.github.io/auto-wcag/'" />    
+  </head>  
+```
+
+```html
+<!-- first valid <meta http-equiv="refresh"> redirects immediately  -->
+<head>
+  <meta http-equiv="refresh" content="0; http://example.com" />
+  <meta http-equiv="refresh" content="5; http://example.com" />
+</head>
+```
+
+```html
+<!-- redirects immediately -->
+<head>
+  <meta http-equiv="refresh" content="0 http://example.com" />
+</head>
+```
+
+### Failed
+
+```html
+<!-- refreshes after 30 seconds -->
+<head>
+<meta http-equiv="refresh" content="30">
+</head>
+```
+
+```html
+<!-- redirects after 30 seconds -->
+<head>
+<meta http-equiv="refresh" content="30; URL='https://auto-wcag.github.io/auto-wcag/'">
+</head>
+```
+
+```html
+<!-- first <meta http-equiv="refresh"> element is not valid, second one redirects after 5 seconds -->
+<head>
+  <meta http-equiv="refresh" content="0: http://example.com" />
+  <meta http-equiv="refresh" content="5; http://example.com" />
+</head>
+```
+
+### Inapplicable
+```html
+<!-- no content attribute -->
+<head>
+<meta http-equiv="refresh">
+</head>
+```
+
+```html
+<!-- no content attribute -->
+<head>
+<meta http-equiv="">
+</head>
+```
+
+```html
+<!-- no http-equiv="refresh" attribute -->
+<head>
+<meta content="30">
+</head>
+```
+
+```html
+<!-- no http-equiv="refresh" or content attribute -->
```

Comment:
If the applicability asks for both, don't we only need a single test case that omits one of the two attributes to cover a scenario where neither are present?